src/rail_gui.cpp

```diff
@@ -1552,6 +1557,25 @@ struct BuildSignalWindow : public PickerWindowBase {
 			size->width = max(size->width, this->sig_sprite_size.width + WD_IMGBTN_LEFT + WD_IMGBTN_RIGHT);
 			size->height = max(size->height, this->sig_sprite_size.height + WD_IMGBTN_TOP + WD_IMGBTN_BOTTOM);
 		}
+
+		if (!_settings_client.gui.show_blocksignals) {
+			switch (widget) {
+				case WID_BS_SEMAPHORE_NORM:
+				case WID_BS_SEMAPHORE_ENTRY:
+				case WID_BS_SEMAPHORE_EXIT:
+				case WID_BS_SEMAPHORE_COMBO:
+				case WID_BS_ELECTRIC_NORM:
+				case WID_BS_ELECTRIC_ENTRY:
+				case WID_BS_ELECTRIC_EXIT:
+				case WID_BS_ELECTRIC_COMBO:
+					*size = Dimension{ 0, 0 };
+					*resize = *size;
+					*fill = *size;
+					break;
+				default:
+					break;
+			}
+		}
```

Comment:
Isn't there a better way to hide widgets? Planes or something?

Reply:
The only planes I've had success finding anything about in OTTD are airplanes, but there may be some corner of NUI that I still haven't discovered exists.

Reply:
Yeah an attempt at wrapping the buttons with an NWID_SELECTION on each row caused massive problems, it's much more trouble than worth to do it "the right way". May as well re-do the entire signals UI then.